tests/test_iotedgedev.py

```diff
@@ -595,42 +434,24 @@ def test_gen_config_with_non_string_placeholder():
     assert "ERROR" not in result.output
 
 
+@mock.patch.dict(os.environ, {"CONTAINER_REGISTRY_SERVER": "localhost:5000"})
 @pytest.mark.skipif(get_docker_os_type() == 'windows', reason='windows container does not support local registry image')
 def test_push_modules_to_local_registry(prepare_solution_with_env):
-    env_container_registry_server = os.getenv("CONTAINER_REGISTRY_SERVER")
     try:
         module_name = "filtermodule"
 
         if module_name in get_all_docker_images():
             remove_docker_image(module_name)
 
-        local_registry = "localhost:5000"
-        envvars.set_envvar("CONTAINER_REGISTRY_SERVER", local_registry)
-
         result = runner_invoke(['push', '-P', get_platform_type()])
-        if result.exit_code == 0:
-            assert 'BUILD COMPLETE' in result.output
-            assert 'PUSH COMPLETE' in result.output
-            assert 'ERROR' not in result.output
-            assert local_registry + "/" + module_name in get_all_docker_images()
-        else:
-            raise Exception(result.stdout)
+
+        assert 'ERROR' not in result.output
+        assert result.exit_code == 0
+        assert 'BUILD COMPLETE' in result.output
+        assert 'PUSH COMPLETE' in result.output
+        assert f"localhost:5000/{module_name in get_all_docker_images()}"
     finally:
-        envvars.set_envvar("CONTAINER_REGISTRY_SERVER", env_container_registry_server)
         if "registry" in get_all_docker_containers():
             remove_docker_container("registry")
         if "registry" in get_all_docker_images():
             remove_docker_image("registry:2")
-
-# # TODO: The output of docker build logs is not captured by pytest, need to capture this before enable this test
```

Comment:
>The output of docker build logs is not captured by pytest [](http://example.com/codeflow?start=10&length=57)

is the root cause for the missing output a debug flag, similar to the issue we found in the deployment e2e tests earlier?

Reply:
I'm not sure what this test is missing, the output of the command does contain some docker output already:
```text
==================================
======== BUILDING MODULES ========
==================================

PROCESSING DOCKERFILE: /workspaces/iotedgedev/tests/assets/test_solution_shared_lib/modules/sample_module/Dockerfile.amd64
BUILDING DOCKER IMAGE: s608d3c36cr.azurecr.io/sample_module:0.0.1-RC-amd64
Step 1/13 : FROM microsoft/dotnet:2.1-sdk AS build-env
 ---> f1c0998716e6
Step 2/13 : COPY ./libs /app/libs
 ---> Using cache
 ---> 54f56272905a
Step 3/13 : COPY ./modules/sample_module/*.csproj /app/modules/sample_module/
 ---> Using cache
 ---> 1b4514da3f81
Step 4/13 : COPY ./modules/sample_module /app/modules/sample_module
 ---> Using cache
 ---> 3d639159cc5f
Step 5/13 : WORKDIR /app/modules/sample_module
 ---> Using cache
 ---> 9db2189989ef
Step 6/13 : RUN dotnet restore
 ---> Using cache
 ---> cbd841a408a5
Step 7/13 : RUN dotnet publish -c Release -o /app/out
 ---> Running in 48d0bd885512
Microsoft (R) Build Engine version 16.2.37902+b5aaefc9f for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 30.41 ms for /app/libs/sharedlib/sharedlib.csproj.
  Restore completed in 253.4 ms for /app/modules/sample_module/sample_module.csproj.
  sharedlib -> /app/libs/sharedlib/bin/Release/netstandard2.0/sharedlib.dll
  sample_module -> /app/modules/sample_module/bin/Release/netcoreapp2.1/sample_module.dll
  sample_module -> /app/out/
 ---> bfc9ed0a1976
Step 8/13 : FROM microsoft/dotnet:2.1-runtime-stretch-slim
 ---> 1bdac099db00
Step 9/13 : WORKDIR /app
 ---> Using cache
 ---> 88710246052f
Step 10/13 : COPY --from=build-env /app/out ./
 ---> Using cache
 ---> 9a6a2f77e152
Step 11/13 : RUN useradd -ms /bin/bash moduleuser
 ---> Using cache
 ---> 645d4ace279a
Step 12/13 : USER moduleuser
 ---> Using cache
 ---> 5191972eb9f0
Step 13/13 : ENTRYPOINT ["dotnet", "sample_module.dll"]
 ---> Using cache
 ---> 5c6d50866ac4
Successfully built 5c6d50866ac4
Successfully tagged s608d3c36cr.azurecr.io/sample_module:0.0.1-RC-amd64
BUILD COMPLETE

PROCESSING DOCKERFILE: /workspaces/iotedgedev/tests/assets/test_solution_shared_lib/sample_module_2/Dockerfile.amd64
BUILDING DOCKER IMAGE: s608d3c36cr.azurecr.io/sample_module_2:0.0.1-RC-amd64
Step 1/13 : FROM microsoft/dotnet:2.1-sdk AS build-env
 ---> f1c0998716e6
Step 2/13 : COPY ./libs /app/libs
 ---> Using cache
 ---> 54f56272905a
Step 3/13 : COPY ./sample_module_2/*.csproj /app/sample_module_2/
 ---> Using cache
 ---> dae479275c91
Step 4/13 : COPY ./sample_module_2 /app/sample_module_2
 ---> Using cache
 ---> abd41bffcb9d
Step 5/13 : WORKDIR /app/sample_module_2
 ---> Using cache
 ---> b66a30a08d69
Step 6/13 : RUN dotnet restore
 ---> Using cache
 ---> a66fbca1e2db
Step 7/13 : RUN dotnet publish -c Release -o /app/out
 ---> Running in ebafcfd39850
Microsoft (R) Build Engine version 16.2.37902+b5aaefc9f for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 32.06 ms for /app/libs/sharedlib/sharedlib.csproj.
  Restore completed in 245.53 ms for /app/sample_module_2/sample_module_2.csproj.
  sharedlib -> /app/libs/sharedlib/bin/Release/netstandard2.0/sharedlib.dll
  sample_module_2 -> /app/sample_module_2/bin/Release/netcoreapp2.1/sample_module_2.dll
  sample_module_2 -> /app/out/
 ---> 4a15928a5d69
Step 8/13 : FROM microsoft/dotnet:2.1-runtime-stretch-slim
 ---> 1bdac099db00
Step 9/13 : WORKDIR /app
 ---> Using cache
 ---> 88710246052f
Step 10/13 : COPY --from=build-env /app/out ./
 ---> Using cache
 ---> 7f2df90ea510
Step 11/13 : RUN useradd -ms /bin/bash moduleuser
 ---> Using cache
 ---> 3a789926a2f2
Step 12/13 : USER moduleuser
 ---> Using cache
 ---> c93c8a5423d8
Step 13/13 : ENTRYPOINT ["dotnet", "sample_module_2.dll"]
 ---> Using cache
 ---> ad0e98bd5053
Successfully built ad0e98bd5053
Successfully tagged s608d3c36cr.azurecr.io/sample_module_2:0.0.1-RC-amd64
BUILD COMPLETE

Expanding image placeholders
Converting createOptions
Deleting template schema version
Expanding \'deployment.template.json\' to \'config/deployment.amd64.json\'
Validating generated deployment manifest config/deployment.amd64.json
Validating schema of deployment manifest.
Deployment manifest schema validation passed.
Start validating createOptions for all modules.
Validating createOptions for module tempSensor
createOptions of module tempSensor validation passed
Validating createOptions for module sample_module
createOptions of module sample_module validation passed
Validating createOptions for module sample_module_2
createOptions of module sample_module_2 validation passed
Validating createOptions for module edgeAgent
createOptions of module edgeAgent validation passed
Validating createOptions for module edgeHub
createOptions of module edgeHub validation passed
Validation for all createOptions passed.
```

Instead of commenting out this test, I'll skip it, so it's at least maintaned

Reply:
I would expect the test to pass with the output you shared.